### ISSUE TITLE
[Submission] 296 params GPT- Sunny Sanyal

### ISSUE THREAD

**Issue by sanyalsunny111** [anadim/AdderBoard#48]
### Author

Sunny Sanyal (@SunnySanyal9)

### Unique Parameter Count

296 params GPT model

### Accuracy

>99% (averaged over 5 seeds)

### Method

Trained (SGD/Adam)

### Architecture

GPT (config): d = n_embd = 4, n_head = 2, n_layer = 1

### Key Tricks

I tried solving this problem with a training recipe approach (curriculm -ish) as most of the submissions here looks like architecture tweaks. Without altering the architecture (much), we modify only the training recipe: the model is pretrained using a curriculum from 2-digit to 10-digit addition and subsequently fine-tuned on the 10-digit task, achieving 99% accuracy. 

<img width="5940" height="4143" alt="Image" src="https://github.com/user-attachments/assets/07b3bb8f-56e4-47b2-a77c-3bd516455886" />

### Link to Code

https://github.com/sanyalsunny111/Smallest-GPT-for-addition

### Verification Output

```shell
All logs and ckpts are provided in the repo.
```

### Additional Notes

Thanks for all your efforts for organizing this game.
